### ISSUE TITLE
Exclude rule on a different rule works although it should not.

### ISSUE THREAD

**Issue by gmponos** [squizlabs/PHP_CodeSniffer#2036]
Hello,

By typo I found out the following:

```
    <rule ref="MyRuleset.Tests.StrictAssertion">
        <include-pattern>*/symfony/tests/Unit/*</include-pattern>
    </rule>

    <rule ref="MyRuleset.PHP.StrictModeInArray.StrictComparison">
        <type>error</type>
        <exclude-pattern>*MyFile*</exclude-pattern>
        <exclude name="MyRuleset.Tests.StrictAssertion.FoundAssertNotEmpty" />
        <exclude name="MyRuleset.Tests.StrictAssertion.FoundAssertNotNull" />
    </rule>
```

You can understand that the rules `MyRuleset.Tests.StrictAssertion.FoundAssertNotEmpty` and `MyRuleset.Tests.StrictAssertion.FoundAssertNotNull` that are excluded should be under the `MyRuleset.Tests.StrictAssertion` and not under `MyRuleset.PHP.StrictModeInArray.StrictComparison`. This never the less made the rules excluded and worked. 

In my understanding it should not work.

At this time rather than reporting this I can't help any further. 😞

**Comment by gsherwood** [comment]
PHPCS will process any `exclude` tag it finds under a `rule` tag.

I could add validation to see if:
1. the rule tag is for a standard or category; and
2. that the exclude rule applies to the standard or category of the rule

But I can't really see the benefit of it.

**Comment by gmponos** [comment]
> But I can't really see the benefit of it.

OK... not my call, I just thought that it's worth mentioning 👍 

You can close it if you wish.

**Comment by monojp** [comment]
I think I just had a similar problem, #3136 might be related too.

We basically have the following
```xml
<rule ref="PSR12"/>

<rule ref="Symfony">
        <exclude name="PSR2"/> <!-- superseded by PSR-12 -->
</rule>
```
and I was expecting that PSR2 will only be excluded in the "scope" of the Symfony rule but apparently all of PSR2 is deactivated, even the includes of PSR2 inside PSR12.

I guess there is no easy fix for this, or workaround, right?

**Comment by rodrigoprimo** [comment]
A similar request has been filed in the new PHP_CodeSniffer repository: PHPCSStandards/PHP_CodeSniffer#1081. Linking for visibility.
